server/app.js

```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
 app.get('/', function(req,res){
     res.redirect('index.html');
 });
- 
-var client = new net.Socket();
+
+require('events').EventEmitter.prototype._maxListeners = 100; //max listeners
+
+var PythonShell = require('python-shell');
+
+var client = new net.Socket();  
+
+client.on('err', function(err){
+            // handle the error safely
+            console.log("Client error");
+            console.log(err)
+        });
 
 io.on('connection',function(socket){ 
+
+    //////////////////////////////////////////////////////////////////
+    // Logging server stuff //
+    var server = net.createServer(function(sock) {
+        sock.on('err', function(err){
+            console.log("vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
+            console.log(err);
+            console.log("\n\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+            server.close();
+            server.listen(5002, '192.168.43.12');  
+        });
+        sock.on('data', function(data) {
+            //console.log(data);
+            //Packet Type
+            var packetType = data.slice(0,1).toString();
+            packetType = "0x" + ascii_to_hexa(packetType);
+            packetType = parseInt(packetType);
+            //console.log("Packet Type: ", packetType);
+            //Payload Length
+            var payloadLength = data.slice(1,2).toString();
+            payloadLength = "0x" + ascii_to_hexa(payloadLength);
+            payloadLength = parseInt(payloadLength);
+            //console.log("Payload Len: ", payloadLength);
+            //Payload
+            var payload = data.slice(2,payload).toString();
+            //console.log(payload);
+            var payload = payload.split(":");
+
+            //console.log("Payload: ", payload[0]);
+            //console.log("Payload: ", payload[1]);
+            //console.log("Payload: ", payload[2]);
+            console.log(packetType , payloadLength, payload);            
+            
+            if(packetType == 0 )
```

Comment:
Should not be `if, if, if, if`, if anything it should be `if, else if`.
Try a switch instead:
```
switch (packetType)
{
    case 0: socket.broadcast.emit; break;
    case 1: socket.broadcast.emit; break;
    case 2: socket.broadcast.emit; break;
    case 3: socket.broadcast.emit; break;
    default: break;
}
```